src/Util/PhpServicesCreator.php

```diff
@@ -0,0 +1,574 @@
+                        $this->addFluentCallMethodNodes($subKey, $serviceValues, $subKey);
+                    }
+                    $this->addLineStmt(';', true);
+                }
+            } elseif (strstr($serviceKey, '.')) {
+                if (isset($serviceValues['class'])) {
+                    $className = $this->addUseStatementIfNecessary($serviceValues['class']);
+
+                    $this->addLineStmt('$services->set(\''.$serviceKey.'\', '.$className.')');
+                } elseif (isset($serviceValues['alias'])) {
+                    $className = $this->addUseStatementIfNecessary($serviceValues['alias']);
+
+                    $this->addLineStmt('$services->alias('.$className.', \''.$serviceKey.'\')');
+                } else {
+                    $this->addLineStmt('$services->set(\''.$serviceKey.'\')');
+                }
+
+                foreach ($serviceValues as $subKey => $subValue) {
+                    if ('class' !== $subKey && 'alias' !== $subKey) {
+                        $this->addFluentCallMethodNodes($subKey, $serviceValues, $subKey);
+                    }
+                }
+                $this->addLineStmt(';', true);
+            } elseif (isset($serviceValues['resource'])) {
+                if (!strstr($serviceKey, '\\')) {
+                    $serviceKey = $serviceValues['namespace'];
+                    unset($serviceValues['namespace']);
+                }
+
+                if ('App\\' === $serviceKey) {
+                    $this->addLineStmt('// makes classes in src/ available to be used as services');
+                    $this->addLineStmt('// this creates a service per class whose id is the fully-qualified class name');
+                }
+
+                if ('App\\Controller\\' === $serviceKey) {
+                    $this->addLineStmt('// controllers are imported separately to make sure services can be injected');
+                    $this->addLineStmt('// as action arguments even if you don\'t extend any base controller class');
+                }
+
+                $this->pushNode($this->factory->methodCall(
+                    new Node\Expr\Variable('services'), 'load', [$serviceKey, $serviceValues['resource']]
+                ));
+
+                foreach ($serviceValues as $subKey => $subValue) {
+                    if ('resource' !== $subKey) {
+                        $this->addFluentCallMethodNodes($subKey, $serviceValues, $subKey);
+                    }
+                }
+
+                $this->addLineStmt(';', true);
+            } elseif ($className = strstr($serviceKey, ' $', true)) {
+                $argument = strstr($serviceKey, '$');
+                $shortClassName = $this->addUseStatementIfNecessary($className).".'".$argument."'";
+                // substr '@'
+                $alias = "'".substr($serviceValues, 1)."'";
+
+                $this->addLineStmt('$services->alias('.$shortClassName.', '.$alias.');', true);
+            }
```

Comment:
After you've refactored to avoid `else`, if you've made it *beyond* the final `if` statement, it will mean that your situation does not match any of our expectations. Should we throw an exception?

Reply:
Yes, but i don't know what to write